cpp/src/ray/ray_config_internal.cc

```diff
@@ -0,0 +1,84 @@
+
+#include "gflags/gflags.h"
+
+#include "ray_config_internal.h"
+
+DEFINE_string(redis_address, "", "The ip address of redis server.");
+
+DEFINE_string(redis_password, "", "The password of redis server.");
+
+DEFINE_string(dynamic_library_path, "", "The local path of the dynamic library.");
+
+DEFINE_string(job_id, "", "Assigned job id.");
+
+DEFINE_int32(node_manager_port, 62665, "The node manager port in Raylet.");
+
+DEFINE_string(raylet_name, "", "Local raylet socket name.");
+
+DEFINE_string(object_store_name, "", "Local object store socket name.");
+
+DEFINE_string(logs_dir, "", "Log dir for workers.");
+
+namespace ray {
+namespace api {
+
+std::shared_ptr<RayConfigInternal> RayConfigInternal::config_ = nullptr;
+
+std::shared_ptr<RayConfigInternal> RayConfigInternal::GetInstance() {
+  if (config_ == nullptr) {
+    config_ = std::make_shared<RayConfigInternal>();
+  }
+  return config_;
+}
+
+void RayConfigInternal::Init(RayConfig &config) {
```

Comment:
Can we simplify this method by making `RayConfigInternal` inherits from `RayConfig`?

Reply:
I think no inherits is good, composite it better than inherits.

Reply:
@kfstorm Some things in RayConfig is not included in RayConfigInternal, such as `address`, `argc_` and `arcv_`.  RayConfig will be transformed. I can add a `RayConfig config` member in `RayConfigInternal` If needed. But now, may it is ok because the config is not very complex.